File: api/schemas/health.py
"""Pydantic schemas for health endpoints."""

from __future__ import annotations

from pydantic import BaseModel


class ServiceStatus(BaseModel):
    """Status of a subsystem."""

    ok: bool
    message: str


class HealthResponse(BaseModel):
    """Health status snapshot."""

    polling_ok: ServiceStatus
    auth_ok: ServiceStatus
    registry_ok: ServiceStatus
    disk_ok: ServiceStatus
    last_updated_at: str
    last_poll_at: str | None = None
    next_poll_at: str | None = None
    poller_status: str = "unknown"
    poll_interval_minutes: int = 0
    error: str | None = None


class PollTriggerResponse(BaseModel):
    """Response from the poll trigger endpoint."""

    status: str

File: webui/src/lib/api/health.ts
import { apiFetch } from './client';

export interface Subsystem {
  ok: boolean;
  message: string;
}

export interface HealthResponse {
  polling_ok: Subsystem;
  auth_ok: Subsystem;
  registry_ok: Subsystem;
  disk_ok: Subsystem;
  last_updated_at?: string;
  last_poll_at?: string | null;
  next_poll_at?: string | null;
  poller_status?: string;
  poll_interval_minutes?: number;
  poll_duration_s?: number | null;
  error: string | null;
}

export interface PollTriggerResponse {
  status: string;
}

export interface PollHistoryEntry {
  day: string;
  duration_s: number;
}

export function getHealth(): Promise<HealthResponse> {
  return apiFetch<HealthResponse>('/api/v1/health');
}

export function triggerPoll(): Promise<PollTriggerResponse> {
  return apiFetch<PollTriggerResponse>('/api/v1/poll/trigger', { method: 'POST' });
}

export function getPollHistory(): Promise<PollHistoryEntry[]> {
  return apiFetch<PollHistoryEntry[]>('/api/v1/health/poll-history');
}
